interfaces/repo.go

```diff
@@ -931,6 +931,14 @@ func (r *Repository) SnapSpecification(securitySystem SecuritySystem, appSet *Sn
 
 	spec := backend.NewSpecification(appSet)
 
+	if opts.AppArmorPrompting {
+		// Need to set UsePromptPrefix() here, but can't, because the backend
+		// might not be apparmor.
+		// And can't wait to do it in spec.AddConnectedPlug, since then it is
+		// re-added for every plug slot, which is innecessary.
+		// XXX: could do it in NewSpecification, but that is called ~750 times.
```

Comment:
I remember adjusting tests when something changed in how we handle specifications and this had to be done for each of the interfaces and each test. Bazillion changes, can relate.

Reply:
Aha, it's the backend-specific `NewSpecification` functions which are called ~750 times, the actual `Backend.NewSpecification` interface is only used a few dozen times :upside_down_face: